kodilla-patterns2/src/main/java/com/kodilla/patterns2/adapter/bookclasifier/MedianAdapter.java

```diff
@@ -4,20 +4,29 @@
 import com.kodilla.patterns2.adapter.bookclasifier.librarya.Classifier;
 import com.kodilla.patterns2.adapter.bookclasifier.libraryb.BookSignature;
 
-import java.util.*;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Set;
 
 public class MedianAdapter extends MedianAdaptee implements Classifier {
 
     @Override
-    public int publicationYearMedian(Set<Book> bookSet) {
+    public int publicationYearMedian( Set<Book> bookSet) {
         Map<BookSignature, com.kodilla.patterns2.adapter.bookclasifier.libraryb.Book> books = new HashMap<>();
+        Iterator<Book> iterator = bookSet.stream()
+                .iterator();
 
-        for( Book book : bookSet) {
-            com.kodilla.patterns2.adapter.bookclasifier.libraryb.Book theBook
-                    = new com.kodilla.patterns2.adapter.bookclasifier.libraryb.Book(book.getAuthor(), book.getTitle(), book.getPublicationYear());
-            books.put(new BookSignature(book.getSignature()), theBook);
+        while( iterator.hasNext()) {
```

Comment:
o matko. pytalem o stream a nie o iteratory - iteratory to stara metodologia.

Reply:
Utknąłem przy wstawianiu do mapy. Nie umiem pobrać signature i book też wywala błąd mimo zmapowania Map<BookSignature, com.kodilla.patterns2.adapter.bookclasifier.libraryb.Book> bookss = bookSet.stream()
                .map(MedianAdapter::mapBookAToBookB)
                .collect(Collectors.toMap(new BookSignature(Book::getSignature), book -> book));